CMSgov/cmcs-eregulations
# EREGCSC-1547 -- New Header

Resolves [EREGCSC-1547](https://jiraent.cms.gov/browse/EREGCSC-1547)

**Description**

New header component created using Vue, not a Django template.

**This pull request changes:**

* Header now a Vue component, used directly in Django templates as well as within the Resources and Search SPAs
* New components use the Vue Composition API, which is available as of Vue 2.7 and in Vue 3. See more [here](https://vuejs.org/guide/extras/composition-api-faq.html#what-is-composition-api)
* Header and Jump To styles have been updated to match [Figma comp](https://www.figma.com/proto/Gx6pz0IxWqGsweFVdezKNe/eRegs-Homepage---Multi-View-Update?page-id=2585%3A11042&node-id=2585-11833&viewport=-991%2C3093%2C0.08&scaling=scale-down&starting-point-node-id=2585%3A11833&show-proto-sidebar=1)
* Some reorganization of Vue components -- for example, JumpTo.vue has been moved into `eregs-vite`.
   - Justification for this move can be found here at [EREGCSC-1782](https://jiraent.cms.gov/browse/EREGCSC-1782) -- the eventual goal is to move all components and api/utility methods into a single project (`eregs-vite`) and export them to be used in that project and any other projects (such as `eregs-component-lib`).  This will make things much easier and reduce quite a bit of duplication.
* Created new `directives` folder to contain custom Vue directives that can be globally imported into both Vite projects.  
   - The `clickaway` directive is now its own file and can be used on any Vue component in either project.
 
**Steps to manually verify this change:**

1. Visit experimental deployment
2. Navigate around site using the header and make sure everything works
